### PR TITLE
[Merged by Bors] - chore(RingTheory/Noetherian): split `Noetherian.lean`

### PR DESCRIPTION
**Pull request by Vierkantor**
This PR splits `RingTheory/Noetherian.lean` into the following files:
* `Noetherian/Defs.lean`: definition of `IsNoetherian` and `IsNoetherianRing`; proof of `isNoetherian_iff`
* `Noetherian/Filter.lean`: ascending chain condition, expressed in terms of filters
* `Noetherian/Orzech.lean`: Orzech's theorem and property
* `Noetherian/Nilpotent.lean`: nilpotent ideals in Noetherian rings
* `Noetherian/UniqueFactorization.lean`: Noetherian implies unique factorization

This change should also clean up the import graph for `LinearAlgebra` somewhat, especially in combination with PR #18731.

Note that there was a timeout in `Topology/Sheaves/Stalks.lean` caused by the instance search order changing. We can either do a local priority change, or simply avoid importing the slow instances. [See also Zulip thread](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Instance.20order.20sensitivity.20in.20.60Topology.2FSheaves.2FStalks.2Elean.60).

---

- [x] depends on: #18762

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Vierkantor** [comment]
Hmm, the import growth is more than I was hoping. Let me see if there's something to split higher up from `Noetherian -> Nilpotent.Lemmas`...

File: Mathlib/RingTheory/Noetherian/Nilpotent.lean [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import Mathlib.RingTheory.Nilpotent.Lemmas
+import Mathlib.RingTheory.Noetherian.Defs
+
+/-!
+# Nilpotent ideals in Noetherian rings
+
+
+## Main results
+
+* `IsNoetherianRing.isNilpotent_nilradical`:
```
Comment (Vierkantor) [inline Mathlib/RingTheory/Noetherian/Nilpotent.lean:15]:
```suggestion
* `IsNoetherianRing.isNilpotent_nilradical`
```

**Comment by Vierkantor** [comment]
Looking into the error, I can't figure out yet what's going on. Compared to master, the synthInstance trace looks exactly the same until the new version runs out of heartbeats. Adding more heartbeats makes everything compile okay. So I assume it's some sort of instance priority changing due to reduced imports. Will check later when I am less sleepy.

**Comment by jcommelin** [comment]
bors d+

**Pull request closed without merging**